service/connection/ClientState.kt

```diff
@@ -119,6 +122,15 @@ class ClientState constructor(
             _client = clientConstructor()
             statusAtomic.set(CONNECTED)
             onSuccess()
+            if (_client?.isCluster == true) {
+                val passwordExpiryDaysOptional: Optional<Long>? = _client?.asCluster()?.user()?.passwordExpiryDays()
+                if (passwordExpiryDaysOptional?.isPresent == true) {
+                    val passwordExpiryDays = passwordExpiryDaysOptional.get()
+                    if (passwordExpiryDays <= PASSWORD_EXPIRY_WARN_DAYS) {
+                        notificationSrv.userWarning(LOGGER, CREDENTIALS_EXPIRE_SOON, passwordExpiryDays)
+                    }
+                }
+            }
```

Comment:
This is fine, but i think you shuold extract it to a method called something like `mayWarnPassworExpiry` - the rest of this method is quite "high level" so you can hide away this implementation detail behind a readable method

Reply:
Agreed, done!